### PR TITLE
impr ARSN-313 upgrade prom-client

### PR DESCRIPTION
**Pull request by naren-rajendran**

### CODE REVIEW

**Review by jonathan-gramain — APPROVED** [review]

File: tests/unit/metrics/ZenkoMetrics.spec.js [hunk 1]
```diff
@@ -94,8 +94,9 @@ describe('ZenkoMetrics', () => {
             'pet_counter{type="puppy"} 2',
         ];
         const lines = {};
-        ZenkoMetrics.asPrometheus().split('\n').forEach(line => {
-            lines[line.trimRight()] = true;
+        const metrics = await ZenkoMetrics.asPrometheus();
```
Comment (rachedbenmustapha) [inline tests/unit/metrics/ZenkoMetrics.spec.js:97]: `ZenkoMetrics.asPrometheus` isn't declared `async`, or is it another one?

File: lib/metrics/ZenkoMetrics.ts [hunk 2]
```diff
@@ -1,26 +1,23 @@
...
-        10000;
-
-promClient.collectDefaultMetrics({ timeout: collectDefaultMetricsIntervalMs });
+// 'timeout' property is not needed/supported
+// https://github.com/siimon/prom-client/blob/199b7d19f8c8c34ee8653264e8dc0e57b420074f/CHANGELOG.md#1200---2020-02-20
+promClient.collectDefaultMetrics();
 
 export default class ZenkoMetrics {
-    static createCounter(params: promClient.CounterConfiguration) {
+    static createCounter(params: any) {
```
Comment (rachedbenmustapha) [inline lib/metrics/ZenkoMetrics.ts]: `unknown` might be more appropriate if type info is not available; why the widening from `CounterConfiguration` to `any`?

Comment (naren-rajendran) [reply lib/metrics/ZenkoMetrics.ts]: The definition has changed in the newer version, I wasn't sure, so used `any`. Modifying it.

File: tests/unit/metrics/ZenkoMetrics.spec.js (hunk 1 above)
Comment (naren-rajendran) [reply tests/unit/metrics/ZenkoMetrics.spec.js:97]: Nope, I missed it.

**Comment by naren-rajendran** [comment]
@bert-e reset

File: lib/metrics/ZenkoMetrics.ts [hunk 3]
```diff
@@ -1,34 +1,31 @@
...
+    static createSummary(params: promClient.SummaryConfiguration<string>) {
         return new promClient.Summary(params);
     }
 
     static getMetric(name: string) {
         return promClient.register.getSingleMetric(name);
     }
 
-    static asPrometheus() {
+    static async asPrometheus() {
```
Comment (rachedbenmustapha) [inline lib/metrics/ZenkoMetrics.ts:28]: If there's nothing to await inside, why make it `async`? Is `promClient.register.metrics()` returning a `Promise`?

Comment (naren-rajendran) [reply lib/metrics/ZenkoMetrics.ts:28]: yes, it is returning a promise

**Review by tmacro — APPROVED** [review]

**Review by alexanderchan-scality — APPROVED** [review]

**Pull request merged**